subxt/src/backend/mod.rs

```diff
@@ -325,9 +325,399 @@ pub enum TransactionStatus<Hash> {
+                            let rpc_params = jsonrpsee::types::Params::new(params.as_deref());
+                            let key: sp_core::Bytes = rpc_params.sequence().next().unwrap();
+                            let value = data.request.pop(key.0);
+                            value.map(|v| serde_json::value::RawValue::from_string(v).unwrap())
+                        }
+                        "chain_getBlockHash" => {
+                            let mut data = self.data.lock().await;
+                            let value = data.request.pop("chain_getBlockHash".into());
+                            value.map(|v| serde_json::value::RawValue::from_string(v).unwrap())
+                        }
+                        _ => todo!(),
+                    }
+                })
+            }
+
+            fn subscribe_raw<'a>(
+                &'a self,
+                _sub: &'a str,
+                _params: Option<Box<serde_json::value::RawValue>>,
+                _unsub: &'a str,
+            ) -> super::rpc::RawRpcFuture<'a, super::rpc::RawRpcSubscription> {
+                Box::pin(async {
+                    let mut data = self.data.lock().await;
+                    let values: RpcResult<Vec<RpcResult<Box<RawValue>>>> =
+                        data.subscription.read().await.map(|v| {
+                            v.into_iter()
+                                .map(|v| {
+                                    v.map(|v| serde_json::value::RawValue::from_string(v).unwrap())
+                                })
+                                .collect::<Vec<RpcResult<Box<RawValue>>>>()
+                        });
+                    values.map(|v| RawRpcSubscription {
+                        stream: futures::stream::iter(v).boxed(),
+                        id: Some("ID".to_string()),
+                    })
+                })
+            }
+        }
+        enum Conf {}
+        impl Config for Conf {
+            type Hash = crate::utils::H256;
+            type AccountId = crate::utils::AccountId32;
+            type Address = crate::utils::MultiAddress<Self::AccountId, ()>;
+            type Signature = crate::utils::MultiSignature;
+            type Hasher = crate::config::substrate::BlakeTwo256;
+            type Header = crate::config::substrate::SubstrateHeader<u32, Self::Hasher>;
+            type ExtrinsicParams = DefaultExtrinsicParams<Self>;
+
+            type AssetId = u32;
+        }
+
+        use crate::backend::Backend;
+
+        fn client_runtime_version(num: u32) -> crate::client::RuntimeVersion {
+            crate::client::RuntimeVersion {
+                spec_version: num,
+                transaction_version: num,
+            }
+        }
+
+        fn runtime_version(num: u32) -> RuntimeVersion {
+            RuntimeVersion {
+                spec_version: num,
+                transaction_version: num,
+                other: HashMap::new(),
+            }
+        }
```

Comment:
This is a good start! I think it would be good to try and make it super clear/simple if possible, eg something along the lines of:

```rust
#[tokio::test]
async fn storage_fetch_values() {
    // Setup
    let mock_data = [
        ("ID1", Ok("Data1".as_bytes())),
        ("ID2", Err(RpcError::DisconnectedWillReconnect("Reconnecting".to_string()))),
        ("ID2", Ok("Data2".as_bytes())),
        ("ID3", Err(RpcError::DisconnectedWillReconnect("Reconnecting".to_string()))),
        ("ID3", Ok("Data3".as_bytes())),
    ];
    let mock_client = mock_storage_client(mock_data.clone());

    // Test
    let backend: LegacyBackend<Conf> = LegacyBackend::builder().build(rpc_client);
    let response = backend
        .storage_fetch_values(
            mock_data.iter().map(|x| x.0.clone().into()).collect(),
            crate::utils::H256::random(),
        )
        .await
        .unwrap();

    let response = response
        .map(|x| x.unwrap())
        .collect::<Vec<StorageResponse>>()
        .await;
    let expected: Vec<_> = mock_data
        .iter()
        .map(|(key, value)| storage_response(key, value))
        .collect();
    assert_eq!(expected, response)
}
```

Reply:
factored some of the common code in the last commit, let me know if it looks good to you